### PR TITLE
changes for making index size configurable in Indexer

### PR DESCRIPTION
**Pull request by hardmettle**
This is to aide https://github.com/ensime/ensime-server/pull/1856
Add configurable parameter for batching while indexing.

### CODE REVIEW

File: core/src/main/resources/application.conf [hunk 1]
```diff
@@ -13,7 +13,7 @@ ensime {
   protocol = "swank"
   exitAfterIndex = false
   disableClassMonitoring = false
-  
+  index.batch.size = 10
```
Comment (fommil) [inline core/src/main/resources/application.conf]: Let's go with index.batchSize

File: core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala [hunk 2]
```diff
@@ -12,9 +12,11 @@ import org.ensime.AkkaBackCompat
 import org.ensime.indexer.SearchService
 import org.ensime.vfs._
 
+
 trait IsolatedSearchServiceFixture extends IsolatedSourceResolverFixture with AkkaBackCompat {
 
   def withSearchService(testCode: (EnsimeConfig, SearchService) => Any)(implicit actorSystem: ActorSystem, vfs: EnsimeVFS): Any = withSourceResolver { (config, resolver) =>
+    import EnsimeConfigFixture.serverConfig
```
Comment (fommil) [inline core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala]: Accidental import?

Comment (fommil) [inline core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala]: One import should be fine

Comment (hardmettle) [inline core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala]: Where to import ? The top level import(as in previous commit wasnt helping). Import in this commit is not working either.

**Comment by hardmettle** [comment]
@fommil cant see what went wrong in you drone ci

**Comment by fommil** [comment]
```diff
Code formatting does not meet the project's standards:
139s
498
diff --git a/core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala b/core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala
139s
499
index 9af3abb..17e6e50 100644
139s
500
--- a/core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala
139s
501
+++ b/core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala
139s
502
@@ -12,8 +12,7 @@ import org.ensime.AkkaBackCompat
139s
503
 import org.ensime.indexer.SearchService
139s
504
 import org.ensime.vfs._
139s
505
 
139s
506
-
139s
507
-trait IsolatedSearchServiceFixture extends IsolatedSourceResolverFixture with AkkaBackCompat with EnsimeConfigFixture{
139s
508
+trait IsolatedSearchServiceFixture extends IsolatedSourceResolverFixture with AkkaBackCompat with EnsimeConfigFixture {
139s
509
 
139s
510
   def withSearchService(testCode: (EnsimeConfig, SearchService) => Any)(implicit actorSystem: ActorSystem, vfs: EnsimeVFS): Any = withSourceResolver { (config, resolver) =>
139s
511
     val searchService = new SearchService(config, resolver)
139s
512
@@ -29,7 +28,7 @@ trait IsolatedSearchServiceFixture extends IsolatedSourceResolverFixture with Ak
139s
513
 
139s
514
 trait SharedSearchServiceFixture
139s
515
     extends SharedEnsimeVFSFixture
139s
516
-    with SharedSourceResolverFixture with EnsimeConfigFixture{
139s
517
+    with SharedSourceResolverFixture with EnsimeConfigFixture {
139s
518
   this: SharedTestKitFixture =>
139s
519
 
139s
520
   private[fixture] var _search: SearchService = _
```

**Comment by fommil** [comment]
once this build finishes it would be interesting to see if changing the number in the `it/application.conf` speeds up the CI in drone. Try 1 and 50 as extreme examples.

**Comment by hardmettle** [comment]
@iravid what went wrong here . It was green earlier.

**Comment by iravid** [comment]
Only the Windows build failed. @fommil said that it's pretty flakey.

File: core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala [hunk 3]
```diff
@@ -12,7 +12,7 @@ import org.ensime.AkkaBackCompat
 import org.ensime.indexer.SearchService
 import org.ensime.vfs._
 
-trait IsolatedSearchServiceFixture extends IsolatedSourceResolverFixture with AkkaBackCompat {
+trait IsolatedSearchServiceFixture extends IsolatedSourceResolverFixture with AkkaBackCompat with EnsimeConfigFixture {
```
Comment (fommil) [inline core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala]: this doesn't look right, this fixture already has an `EnsimeConfig` injected... all you need to do is put

```
import EnsimeConfigFixture.serverConfig
```

in the imports of any file that creates a `SearchService`

File: core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala [hunk 4]
```diff
@@ -12,7 +12,6 @@ import org.ensime.AkkaBackCompat
 import org.ensime.indexer.SearchService
 import org.ensime.vfs._
 
-
```
Comment (fommil) [inline core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala]: if you compile the project with sbt it will auto-reformat the code.

**Comment by hardmettle** [comment]
@fommil uggh .. I got confused ? How will this go green?

File: core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala (hunk 2 above)
Comment (fommil) [reply core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala]: just remove this line and it all compiles

File: core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala [hunk 5]
```diff
@@ -28,7 +29,7 @@ trait IsolatedSearchServiceFixture extends IsolatedSourceResolverFixture with Ak
 
 trait SharedSearchServiceFixture
     extends SharedEnsimeVFSFixture
-    with SharedSourceResolverFixture {
+    with SharedSourceResolverFixture with EnsimeConfigFixture {
```
Comment (fommil) [inline core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala]: this `with EnsimeConfigFixture` is not needed

File: core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala [hunk 6]
```diff
@@ -15,6 +15,7 @@ import org.ensime.vfs._
 trait IsolatedSearchServiceFixture extends IsolatedSourceResolverFixture with AkkaBackCompat {
 
   def withSearchService(testCode: (EnsimeConfig, SearchService) => Any)(implicit actorSystem: ActorSystem, vfs: EnsimeVFS): Any = withSourceResolver { (config, resolver) =>
+    import EnsimeConfigFixture.serverConfig
```
Comment (fommil) [inline core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala]: just remove this line

**Comment by hardmettle** [comment]
@fommil is this ok ? Or do we need the travis to go green as well?

**Comment by fommil** [comment]
You need to remove the other line too.

**Pull request merged**